### DISCUSSION TITLE
DEFINE COLUMN statement not working with a tabular model coming from Excel (current 365 version)

### DISCUSSION THREAD

**Discussion by RainerHollmannGH** [DaxStudio/DaxStudio#1198] [Q&A]
I want to develop a new calculated column for an tabular model coming from excel. Like writing and testing of measures via a dax query in dax studio I want to do the same with the new calculated column. But I received in error message that the "test" - column could not found in the table or is not allowed in this expression? What's strange is that I can't find any reference to a possible restriction (=not working with excel) anywhere (not even in https://www.sqlbi.com/). Any idea?

DEFINE
COLUMN PO[test] = 1 /* of course the table PO is available in the model and the model is connected */

**Comment by dgosbell** [comment]
We cannot see how you are calling this column and that is probably where the issue is. I ran the following simple query against an Excel model and it seemed to work.
```
DEFINE
    COLUMN company[test] = 1
EVALUATE
VALUES ( company[test] )
```

![image](https://github.com/DaxStudio/DaxStudio/assets/8241830/2a16f6cc-9a15-4a89-a209-a57712ba7091)

But note that the tabular engine in Excel is an older version compared to the one in Power BI and as such often does not support all the latest functionality. Also note that as per the caution in the parameters section of the documentation [here](https://learn.microsoft.com/en-us/dax/define-statement-dax) the use of query scoped columns like this is intended for internal use only. So it may not work as expected in all situations. As such I would not recommend test calc columns in this manner. You are better off just creating them in the Excel interface.

**Reply by RainerHollmannGH** [reply]
OK, first of all, thank you very much for the quick answer and your test. It's really interesting that it's running on your side. So, apart from my complex model, I created a simple one in Excel with a single table and tried again. Unfortunately same error (see screencopy attached). Maybe it's due to my versions Excel (Microsoft® Excel® for Microsoft 365 MSO (Version 2310 Build 16.0.16924.20054) 64-bit)/DAX analyzer (3.0.10)?
Regardless of whether I could still get it to work for me, I would of course be happy to take your tip with me and then test the calculated columns in Excel (as before), - which is of course slow/tough because the corresponding reports are then updated directly on every small change (in my case: many reports, long time waiting....)
![dax studio define column with excel model error](https://github.com/DaxStudio/DaxStudio/assets/119947604/27cd3a0f-b68a-420f-a27b-7f4a7f03fd7c)
